engine/src/main/java/org/hibernate/search/backend/impl/lucene/IndexWriterDriver.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.impl.lucene;
+
+import java.io.IOException;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import org.apache.lucene.document.Document;
+import org.apache.lucene.index.IndexWriter;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.Query;
+import org.hibernate.search.backend.impl.lucene.analysis.ConcurrentlyMutableAnalyzer;
+import org.hibernate.search.util.impl.ScopedAnalyzer;
+
+/**
+ * Encapsulates various operations to be performed on a single IndexWriter.
+ * Avoid using {@link Workspace#getIndexWriter()} directly as it bypasses lifecycle
+ * management of the IndexWriter such as reference counting, potentially leading to leaks.
+ *
+ * @author Sanne Grinovero <sanne@hibernate.org> (C) 2015 Red Hat Inc.
+ */
+public final class IndexWriterDriver {
```

Comment:
Ok, that's the same name I thought about too but I then changed it to `xDriver` as it does a bit more complex _driving_ of the IW. I'll change it to your suggestion, the Driver aspect is probably too unusual / flashy.